sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -0,0 +1,982 @@
+            string topicName,
+            string subscriptionName,
+            TokenCredential credential,
+            ServiceBusProcessorClientOptions clientOptions = default)
+            : this(new ServiceBusConnection(fullyQualifiedNamespace, GetSubscriptionPath(topicName, subscriptionName), credential, clientOptions?.ConnectionOptions), clientOptions?.Clone() ?? new ServiceBusProcessorClientOptions())
+        {
+        }
+
+        private static string GetSubscriptionPath(string topicName, string subscriptionName)
+        {
+            return $"{topicName}/{subscriptionName}";
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusProcessorClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connection">The <see cref="ServiceBusConnection" /> connection to use for communication with the Service Bus service.</param>
+        /// <param name="clientOptions">A set of options to apply when configuring the consumer.</param>
+        ///
+        public ServiceBusProcessorClient(
+            ServiceBusConnection connection,
+            ServiceBusProcessorClientOptions clientOptions = default)
+        {
+            _receiver = new ServiceBusReceiverClient(connection, clientOptions?.ToServiceBusReceiverClientOptions() ?? new ServiceBusReceiverClientOptions());
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusProcessorClient"/> class.
+        /// </summary>
+        ///
+        protected ServiceBusProcessorClient()
+        {
+        }
+
+        /// <summary>Indicates that the receiver wants to defer the processing for the message.</summary>
+        /// <param name="lockToken">The lock token of the <see cref="ServiceBusMessage" />.</param>
+        /// <param name="propertiesToModify">The properties of the message to modify while deferring the message.</param>
+        /// <param name="cancellationToken"></param>
+        /// <remarks>
+        /// A lock token can be found in <see cref="ServiceBusMessage.SystemPropertiesCollection.LockToken"/>,
+        /// only when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
+        /// In order to receive this message again in the future, you will need to save the <see cref="ServiceBusMessage.SystemPropertiesCollection.SequenceNumber"/>
+        /// and receive it using <see cref="ServiceBusReceiverClient.ReceiveDeferredMessageAsync(long, CancellationToken)"/>.
+        /// Deferring messages does not impact message's expiration, meaning that deferred messages can still expire.
+        /// This operation can only be performed on messages that were received by this receiver.
+        /// </remarks>
+        public virtual async Task DeferAsync(string lockToken, IDictionary<string, object> propertiesToModify = null,
+            CancellationToken cancellationToken = default)
+        {
+            // TODO implement
```

Comment:
We have all the settlement methods(complete, abandon, defer, deadletter) implemented in `ServiceBusReceiverClient`. Can we just copy the same implementation here?

Reply:
Yeah will do this in separate PR.